lib/chef/mixin/train_helpers.rb

```diff
@@ -0,0 +1,44 @@
+#--
+# Author:: Lamont Granquist <lamont@chef.io>
+# Copyright:: Copyright 2010-2018, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "stringio" unless defined?(StringIO)
+
+class Chef
+  module Mixin
+    module TrainHelpers
+      def file_exist?(*args)
+        if Chef::Config.target_mode?
+          Chef.run_context.transport_connection.file(args[0]).exist?
+        else
+          File.exist?(*args)
+        end
+      end
+
+      # XXX: modifications to the StringIO won't get written back
+      def file_open(*args, &block)
+        if Chef::Config.target_mode?
+          content = Chef.run_context.transport_connection.file(args[0]).content
```

Comment:
do we need any error handling here if the file can't be read or are we fine with just letting train blow up with its own error?

Reply:
doesn't matter right now, the only place we use it we check file_exist? beforehand.

that's problematic, because what should we do, and should we just fake out standard File.open errors so that code doesn't have to change?  or should we throw some other Chef::Exception error?  or should we just require consumer code to catch the error(s) that train throws?

and this whole approach is super hacky and i don't know how far it'll actually go, but i can't sit down and design the correct thing without doing exploratory surgery.